src/main/scala/cromwell/engine/backend/BackendCallDescriptor.scala

```diff
@@ -0,0 +1,29 @@
+package cromwell.engine.backend
+
+import cromwell.engine.{AbortRegistrationFunction, WorkflowDescriptor}
+import cromwell.engine.workflow.{FinalCallKey, CallKey, BackendCallKey}
+import wdl4s._
+import wdl4s.values.WdlValue
+
+
+sealed trait CallDescriptor[K <: CallKey] {
+  def workflowDescriptor: WorkflowDescriptor
+  def key: K
+  def locallyQualifiedInputs: CallInputs
+}
+
+final case class BackendCallDescriptor(workflowDescriptor: WorkflowDescriptor,
```

Comment:
Also note that this demonstrates how when implementing a scala trait you can always refine downward (def -> lazy val -> val -> var) for your concrete instantiation as the implementation is using a val an the trait specified a def